### ISSUE TITLE
Photos layout for link shares

### ISSUE THREAD

**Issue by grenzverkehr** [nextcloud/photos#236]
It should be possible to present photos via a public photo link in the photo gallery view. This is important for everyone who values the presentation of their photos. Up to now only the files view can be shared publicly. 
This was possible in the Gallery app: instead of "/s/name" (Files view), the URLs were then called "apps/gallery/s/name" (Gallery view).

**Comment by dsiminiuk** [comment]
Without the gallery view from a shared URL this plugin I don't see the value of this plugin. I may as well share the folder the regular way.

hint: The old Gallery app was good at this.

**Comment by skjnldsv** [comment]
@nickvergessen we need to add the redirect in server, right?

**Comment by nickvergessen** [comment]
Well the gallery way was wrong and one reason for starting new. It duplicated all the share controller logic into the app to have a second URL.

From my POV we should just make sure the photos app (same with other viewer) show on public shares.

**Comment by skjnldsv** [comment]
> From my POV we should just make sure the photos app (same with other viewer) show on public shares.

Well, we have viewer on public page now, but not sure about having the photo view there :thinking: 
Thoughts @jancborchardt ?

**Comment by diamaunt** [comment]
I can't possibly emphasize this issue enough.
the photos is completely useless as it is, the entire reason for using it is to be able to share photo galleries with others. 

Please fix this as soon as you can!

**Comment by diamaunt** [comment]
@nickvergessen the old gallery app may have been "wrong", but at least it *worked*. 

I'd love to love photos, but right now, it's just broken my site.

**Comment by dsiminiuk** [comment]
Yup I commented on that thread earlier.

Tiny screen buttoons, giant fingers...
________________________________

**Comment by nickvergessen** [comment]
> the photos is completely useless as it is

Way to motivate people... It might not cover your usecase (yet). But the new Photos/Album view is really helping me and also other people seem to like it.

> the old gallery app may have been "wrong", but at least it worked.

Again, that is your usecase, not everyones. The Gallery used it's own endpoint, previews and logic, which even leads to security issues like showing all images of folders which are shared as "files drop" (upload only): https://nextcloud.com/security/advisory/?id=NC-SA-2019-012
So we had to get rid of the gallery and replace it with something new. We see that it is not covering all usecases yet, that where covered by the gallery app, but no one claims it does and no one is saying it's not going to change in the future. But maybe this is your time to shine @diamaunt with your empty github account to start contributing. See the readme how to get involved: https://github.com/nextcloud/photos#maintainers

And if you can't code, you can do other things that take away load from @skjnldsv so they can find the time to code stuff like this.

**Comment by Boki4d** [comment]
@nickvergessen 
I can't code, but I really need the the "gallery" view for the shared links. How can I help? 

Guess I need to learn to code... :/

**Comment by diamaunt** [comment]
> > the photos is completely useless as it is

> Way to motivate people... 
But maybe this is your time to shine @diamaunt with your empty github account

I see you're adept at motivating people yourself, by attacking THEM, instead of the issue, well done.

WELL, I'm certainly motivated.

In all fairness, I should have said "The Photos app is completely useless *TO ME* as it is.   Would that have avoided your rant?  (couldn't the time spent belittling me have been used more productively?  I didn't degrade anyone, I didn't attack anyone, I didn't belittle anyone.   Who do you think you are? Linus Torvalds?)

> >    Please fix this as soon as you can!

> no one is saying it's not going to change in the future.

And, if you'll note, that was *exactly* what I was *asking* for, I even said "please". 

> And if you can't code, you can do other things that take away load from @skjnldsv so they can find the time to code stuff like this.

Sure. what does @skjnldsv need?

**Comment by skjnldsv** [comment]
> Sure. what does @skjnldsv need?

Time. Anyone willing to dive into this please do. Feel free to come and ask questions. We'll happily help anyone.

@diamaunt, @nickvergessen isn't the only one thinking what he wrote.
Please have a look at our code of conduct https://nextcloud.com/contribute/code-of-conduct/

> In all fairness, I should have said "The Photos app is completely useless _TO ME_ as it is. Would that have avoided your rant?

No, lors of people already discussed this here or in others issues, we do not want additional feedback stating if you dislike it or not. Please use Github's reaction feature as a way to advocate for already expressed feedback.

What you could have said that would have motivated people would be: 

> Hi! I'd also really like to see this feature implemented, what can I do to help?
> Cheers
> Michael

Thanks for your motivation, please keep that in mind next time you're participating to an open source project :)
Take care

**Comment by skjnldsv** [comment]
@Boki4d thank for your interest!
We could use a proper definition of what this issue is about.
There is currently the `files list view` in folder sharing and the `grid view`.
The `grid view` is pretty close to what we already have in the `albums view` here in photos. What seems to be missing would be the timeline.

What would you like to see as a third sharing option then?
The album view with browsable folders or the **unified** view like the timeline? :)

**Comment by diamaunt** [comment]
If I, within nextcloud, go into the photos app, it is quite nice. I can click on an album, and then on a photo, and it displays the photo quite nicely, and allows for a slideshow.

However, I *cannot* share that *view*.  All that can be *shared* is a *file list*, which does not allow *viewing* of photos, only *downloading* of them.

I want to be able to share the photo viewing/slideshow with others, via a share link, so that I can put that link on a website and share photos of events that our club has had.

In other words, it appears that Photos is a purely internal utility, which is fine for the files owner, but denied to everybody else.

**Comment by skjnldsv** [comment]
> I want to be able to share the photo viewing/slideshow with others

This was already fixed and part of 18.0.3. See forum and other dedicated issues: https://github.com/nextcloud/photos/issues/128

**Comment by Boki4d** [comment]
> > I want to be able to share the photo viewing/slideshow with others
> 
> This was already fixed and part of 18.0.3. See forum and other dedicated issues: #128 
I think that would be the solution for most of us, so it's nice that it is already planned for 18.0.3. :)

**Comment by skjnldsv** [comment]
Then this discussion is about the photos layout in public shares :)

**Comment by diamaunt** [comment]
I don't know about what's in 18.0.3 because apparently, it's not available?

But it's not the *layout*, it's that there is no photos at all in public shares, at this time.

Are you saying that photos can be made *available* in public shares in 18.0.3?

**Comment by skjnldsv** [comment]
> I don't know about what's in 18.0.3 because apparently, it's not available?

This is the incoming release. Please read the other issue #128!



> But it's not the _layout_, it's that there is no photos at all in public shares, at this time.
> Are you saying that photos can be made _available_ in public shares in 18.0.3?

This issue IS about the layout. You can already share folders like it was always possible on nextcloud. Photos are files, so you can share photos.
Please refer to the documentation and use manual if you're new to Nextcloud.

I will not waste any more time discussing irrelevance to this issue. Thanks for your understanding.

**Comment by Boki4d** [comment]
> But it's not the _layout_, it's that there is no photos at all in public shares, at this time.
> 
> Are you saying that photos can be made _available_ in public shares in 18.0.3?

I'm running 18.0.2 and when I share a folder via link, the pictures are displayed in the file view and can be downloaded. The only problem is, that you can't start a slideshow.

**Comment by diamaunt** [comment]
@Boki4d I don't want people to have to *download* a photo, I want them to be able to *see* the photo, just like nextcloud can bring up a PDF, or a xlsx or a docx file, and display the contents, I want it to once again be able to *display* the contents of a photo when someone clicks on it.

I have galleries of our club activities, I want people to be able to see them, without downloading hundreds of megs of pictures.

I know that this will be 'off topic' but I wanted you to understand.

**Comment by Boki4d** [comment]
> I know that this will be 'off topic' but I wanted you to understand.
There's a big difference between "photos are there, but have to be downloaded instead of beeing displayed in a slideshow" and "there is no photos at all in public shares".

As far as I understood, the slideshow-feature will return with 18.0.3, which means that the guys from nextcloud DID understand the issue and are working on it.

**Comment by dsiminiuk** [comment]
> > I want to be able to share the photo viewing/slideshow with others
> 
> This was already fixed and part of 18.0.3. See forum and other dedicated issues: #128 https://help.nextcloud.com/t/new-photo-app-in-nextcloud-18/69949/40?u=skjnldsv
> This want aims to discuss the portage of the photos layout to link shares as well. :)

Updated to 18.0.3. No change in the photo tile view from a shared URL that I can tell. :(

**Comment by nickvergessen** [comment]
18.0.3 was an intermediate security release. It will come with 18.0.4

**Comment by Leon5x** [comment]
A workaround until NC 18.0.4 is released on 23 April is described here: 
[https://help.nextcloud.com/t/new-photo-app-in-nextcloud-18/69949/20](https://help.nextcloud.com/t/new-photo-app-in-nextcloud-18/69949/20)
You can enable the old gallery app under NC 18. But this app does only work on public shares without a password when you use the sharing link like before (url.com/apps/gallery/s/awhdkawhd).
Btw, thanks for fixing everything for the next release properly!

**Comment by maltokyo** [comment]
@skjnldsv you mentioned above that this was fixed in 18.0.3, which is what I am running.  It is not fixed for me, any hints or tips to get it working?  I upgraded from 17, and my users are screaming at me ;(

**Comment by jancborchardt** [comment]
> > From my POV we should just make sure the photos app (same with other viewer) show on public shares.
> 
> Well, we have viewer on public page now, but not sure about having the photo view there thinking
> Thoughts @jancborchardt ?

If you share any folder from the Photos app, the view that others should get is exactly as you see the photos, in a nice photo view. :)

**Comment by derhagen** [comment]
> @Boki4d thank for your interest!

Until now, I used public gallery links to send my relatives photos of my vacations. The semi-large thumbnails easily allowed to browse and explore the photos online without having to download them first. I agree, that the grid view does basically the same thing. Nevertheless, I would love to have the option to define a default view (and possibly default ordering). Otherwise I'll have to tell my grandparents 'click on this link and then click the tiny button in the upper right corner to see my photos'.

I can imagine three possiblities for an easy fix for my use case:
- If a shared folder contains only photos, show grid view by default
- Allow a GET parameter ?view=grid to open grid view by default
- Allow to set default viewing options while creating the shared link (probably not so trivial but more beautiful and more extensible)

As a plus, reintroducing the before-mentioned timeline view would be great. Also, being able to control the default ordering in the same way as I proposed for the view, would make sense.

If you have a look at Google Photos and the many options to present shared albums, this is probably what I would expect from an app like this as a gold standard, but I guess this is way out of scope for this issue.

**Comment by nautik1** [comment]
Thank you @Leon5x for the workaround, reactivating Gallery saved the day for me too.
And glad to see a solution coming in NC 18.0.4 (on 23 April you said?) without bringing back a deprecated app.

Allowing anyone to view and slide through the pictures I share with them using a link, without having to download each picture, is pretty much the **_only thing_** I asked of the previous Gallery app (and will ask of the new photo app)...

**Comment by diamaunt** [comment]
Politely echoing the comments of @nautik1.

**Comment by maltokyo** [comment]
I also deactivated "photos" and installed the "gallery" app. It's great now! I can live with this for as long as it takes. Photos is not usable at all for me.

**Comment by dsiminiuk** [comment]
Updated NC to 18.0.4 today, the Photos app is still v1.0.0

Must I clone and install manually?

I'm hoping the shared URL for views (a la Gallery) is working.

**Comment by skjnldsv** [comment]
> I'm hoping the shared URL for views (a la Gallery) is working.

Well, as you can see, this issue is not closed and is labelled "to develop".

**Comment by dsiminiuk** [comment]
I was pretty certain I'd read here it was going to do that in 18.0.4.

In the meantime it provides no value to me (and others I know personally here)... disabled.

**Comment by skjnldsv** [comment]
> > I want to be able to share the photo viewing/slideshow with others
> 
> This was already fixed and part of 18.0.3. See forum and other dedicated issues: #128 [help.nextcloud.com/t/new-photo-app-in-nextcloud-18/69949/40?u=skjnldsv](https://help.nextcloud.com/t/new-photo-app-in-nextcloud-18/69949/40?u=skjnldsv)
> This want aims to discuss the portage of the photos layout to link shares as well. :)

See above

**Comment by Boki4d** [comment]
> I was pretty certain I'd read here it was going to do that in 18.0.4.

I'm pretty sure this was related to the preview of pictures without having to download them.

**Comment by dsiminiuk** [comment]
If you read every post on this issue (#236) it is exactly about sharing a folder and the thumbnail view in the same manner as the Gallery app did, for vacations and baby pictures so my computer illiterate mother-in-law of 89 years can see what I intend her to see. Otherwise why have a NC server?

I see above it was mentioned that the requested enhancement was to be in 18.0.3 but it wasn't, and then it was expected in 18.0.4, but it isn't.

I appreciate very much that people have skills I don't possess to do what I cannot, and also volunteer their time  and knowledge to enhance this software, but I don't set the expectations. There is either a gap in understanding of the requirements or a gap in expectations and it seems to me to be the latter.

On the Gallery github page it says "Gallery has been replaced by Photos". I beg to differ.

This is just my (and perhaps others) perspective, as a user. </rant>

**Comment by maltokyo** [comment]
I upgraded. Hoped photos would be usable. Found it still isn't near what we had in gallery and reenabled gallery. 

It's quite puzzling how gallery was replaced by something far inferior.

**Comment by derhagen** [comment]
@dsiminiuk As @Boki4d explained and as it is explicitly mentioned by @skjnldsv, the issue that was fixed in 18.0.3 was #128, which is about the slideshow functionality. You will find this one working in your 18.0.4 installation. As it was said before, this issue is "Open" and "to develop". That means, the issue is **not fixed yet** but somebody will eventually take care of it.

It is not particularly helpful to step into an open issue and complain that it is not fixed yet. A workaround to install the gallery app has been published, so nobody forces you to use the photos app, for now.

**Comment by skjnldsv** [comment]
> so my computer illiterate mother-in-law of 89 years can see what I intend her to see. Otherwise why have a NC server?

I'm pretty sure most people using NC are not here because they're sharing pictures with their 89 years-old illiterate grandma :woman_shrugging: 
___
Last reminder before I lock this topic, everything have been thoroughly discussed already. If you're just here to complain, you can pass by and go somewhere else. Thank you.

**Comment by GitWidi** [comment]
@derhagen Thanks for your engagement here. I fully support your opinion.
I have also used the gallery app as an electronic photo album so far. We organize mountain tours with changing groups of participants. After a tour I collected the photos (shared link with write permission), organized them in a photo album and sent the link of the photo album by mail to the participants, most of whom have no account in the cloud.  
I love the possibility to customize the design of the photo album in the file gallery.cnf. Also, I often have panoramic shots that are not taken into account in the grid display of the photos app. The panorama image is reduced to a small, square thumbnail which shows only an insufficient part of the whole image. For me the grid view is just a bad workaround compared to that was already possible in the gallery app. 

@skjnldsv Yes, the most people using NC would not need this feature. But for some users, this change from NC 17 to NC 18 means a significant step backwards in the usability of Nextcloud.
For me, this is a key feature. Therefore I still hesiate the upgrade to NC 18. If there is no other way, I hope I can still disable the fotos app and install the gallery app instead. Of course, it would be much nicer if this function was included in the standard installation again.

**Comment by skjnldsv** [comment]
> @skjnldsv Yes, the most people using NC would not need this feature.

I never said this, I can completely understand that lots os uses would like this feature in :)
It's labelled as `to develop`, so we agree :wink:

**Comment by maltokyo** [comment]
What I don't understand is, why not just bring back the gallery app? I use it via the hack already posted. But why not bring it back officially? (Until photos does at least the same thing)

**Comment by skjnldsv** [comment]
> everything have been thoroughly discussed already.

Like said, check forum, check other issues. Been answered multiple times

**Comment by maltokyo** [comment]
I did search. Can't find any valid reasoning for doing that. Care to share?

**Comment by Boki4d** [comment]
> I did search. Can't find any valid reasoning for doing that. Care to share?
As far as I remember, the old Gallery app was getting harder and harder to maintain and would have needed a major rework. So they made a decision to ditch it in favor of the new Photos app.
Unfortunately I can't find the offical post, so it's only a memory log without any claim to completeness/correctness.

**Comment by maltokyo** [comment]
I get the need to replace it. Makes fully sense if it's hard to maintain. I don't think anyone opposes that. The only ask from many here is to leave gallery in place until that replacement app is ready. No need to update and maintain gallery. Just leave it working how it was(!) - the reason why that's not possible, I don't understand.

**Comment by ToneLune** [comment]
Hello,

Apparently, the problem has been resolved since version 18.0.4. For me, when I want to display the preview of a photo in a shared folder: the preview loads but the photo never appears with this error in the logs:

```
[webdav] Debug: Sabre\DAV\Exception\NotAuthenticated:  at <<closure>>

0. /home/nextcloud/www/apps/dav/lib/Connector/Sabre/ServerFactory.php line 148
   {closure}("*** sensitive parameters replaced ***")
1. <<closure>>
   OCA\DAV\Connector\Sabre\ServerFactory->OCA\DAV\Connector\Sabre\{closure}("*** sensitive parameters replaced ***")
2. /home/nextcloud/www/3rdparty/sabre/event/lib/EventEmitterTrait.php line 105
   call_user_func_array(Closure {}, ["*** sensitive  ... "])
3. /home/nextcloud/www/3rdparty/sabre/dav/lib/DAV/Server.php line 466
   Sabre\Event\EventEmitter->emit("beforeMethod", ["*** sensitive  ... "])
4. /home/nextcloud/www/3rdparty/sabre/dav/lib/DAV/Server.php line 254
   Sabre\DAV\Server->invokeMethod("*** sensitive parameter replaced ***", "*** sensitive parameter replaced ***")
5. /home/nextcloud/www/apps/dav/appinfo/v1/publicwebdav.php line 109
   Sabre\DAV\Server->exec()
6. /home/nextcloud/www/public.php line 81
   require_once("/home/nextcloud ... p")

PROPFIND /public.php/webdav/DSC03980.JPG

[no app in context] Debug: Deprecated event type for OCP\IPreview:PreviewRequested: Symfony\Component\EventDispatcher\GenericEvent

GET /index.php/apps/files_sharing/publicpreview/cMyNrcQqsejtsdN?fileId=254569&file=%2FDSC04014.JPG&c=a9d10f5416983ec20f3a5b3e33d5647c&x=250&y=250
from xx.xxx.xxx.xx at 2020-04-29T20:29:36+00:00
```

Any idea ?
Thanks !

**Comment by skjnldsv** [comment]
@ToneLune please open your own issue

**Comment by jancborchardt** [comment]
Locking this issue to contributors only cause of too much irrelevant off-topic discussion.

Clear and concise design proposal still stands from my comment above at https://github.com/nextcloud/photos/issues/236#issuecomment-606550812

**Comment by jospoortvliet** [comment]
> I get the need to replace it. Makes fully sense if it's hard to maintain. I don't think anyone opposes that. The only ask from many here is to leave gallery in place until that replacement app is ready. No need to update and maintain gallery. Just leave it working how it was(!) - the reason why that's not possible, I don't understand.

It had security issues, a huge performance issue for large instances and other problems, so while it might be 'good enough' and even 'better than photos' for some, it was worse than photos for many others. That's the problem of course with every time you write a new app to replace an old one, you will always do somethings better and some things worse. We try to fix the things we do better and keep what we did worse, but it will be a long time between 'it is better for most people' (where I think it was with 18) and 'it is better for EVERYONE'.

Now it would've been better if we had made the old gallery app available in the app store, agreed... Then again, that would have been more work, especially as we would have to fix at least the security problem, meaning less time to make Photos better.

Developing software is about trade-offs and often you can't just make everyone happy. Luckily Nextcloud is open source, so one solution for that is that everyone who is unhappy helps fix or add the things they are missing. But if they are unwilling or unable to do that, then yes, there is not much to do.

**Comment by szaimen** [comment]
This is now possible via the albums feature which allows to share pictures via a link. So I'd say this is fixed?

**Comment by joshtrichards** [comment]
Implemented awhile ago
